packages/api/cms-api/src/auth/guards/comet.guard.ts

```diff
@@ -45,18 +42,7 @@ export function createCometAuthGuard(type?: string | string[]): Type<IAuthGuard>
             }
 
             const canActivate = await super.canActivate(context);
-            const isAllowed = isObservable(canActivate) ? await lastValueFrom(canActivate) : canActivate;
-
-            const roles = this.reflector.getAllAndOverride<string[]>(allowForRoleMetadataKey, [context.getHandler(), context.getClass()]) ?? [];
-            if (isAllowed && roles.length > 0) {
-                const userRole = ((this.getRequest(context).user as CurrentUserInterface) || undefined)?.role;
-                if (!userRole) return false;
-
-                const userRoleIsAllowed = roles.some((role) => role.toLowerCase() === userRole.toLowerCase());
-                return userRoleIsAllowed;
-            }
-
-            return isAllowed;
+            return isObservable(canActivate) ? lastValueFrom(canActivate) : canActivate;
```

Comment:
Why not? It still checks for `disableGlobalGuard`- and `publicApi`-decorators.

Reply:
Because the checks for disalbeGlobalGuard und publicApi where needed to disable the guard, and the actual guard (that checks for allowForRole) got removed

Reply:
If an application chooses not to use the UserPermission-module it might still want a guard which supports these decorators. Furthermore this guards provides support for [GraphQL](https://docs.nestjs.com/recipes/passport#graphql).

Reply:
Ok, now I understand.

I thought the purpose of this guard is to support resolvers (+quers/controllers) decorated with `@AllowForRole` - and that got removed.

But it also checks for (any) authed user.

Not a good name actually....

Reply:
CometAuthGuard -> CommonAuthGuard?
CometAuthGuard -> GlobalAuthGuard?

I don't know. Suggestions?

Reply:
- `CometAuthGuard` extends `@nestjs/passport` `AuthGuard`
- it enforces an authed user (does passport AuthGuard this too?)
- it respects disableGlobalGuard and publicApi decorators

---

- maybe `PassportAuthGuard`? (not really more meaningful)
- or: move the whole guard into the application where `AuthGuard` is a perfect name (and provide helpers for accessing the decorators)

Reply:
imho: `CometAuthGuard` > `PassportAuthGuard`
The idea of moving the guard into the application sounds good initially but seems not really practical to me. Calling helpers in severals methods doesn't improve much.

Reply:
Question: does the out-of-the-box @nestjs/passwort AuthGuard enforce a user?